### DISCUSSION TITLE
Doing a relative navigation and then going back to previous page freezes all navigation on Prism 9.0.271-pre

### DISCUSSION THREAD

**Discussion by SandipAhluwalia** [PrismLibrary/Prism#3074] [.NET MAUI]
So I have just updated our Xamarin.Forms app's Prism.Unity.Forms package to v9.0.271-pre (before I migrate to Prism.Dryloc.Maui to start the MAUI migration), and noticed the relative navigation becomes broken/frozen after tapping the back button on the existing Xamarin.Forms app, so as example

Starting from "NavigationPage/HomePage", I do something like the following:

```
var navParameters = new NavigationParameters();
navParameters.Add("myBooking", selectedBooking);
await m_navigationService.NavigateAsync("MyDetailsPage", navParameters, false);
```

Then from MyDetailsPage I go back to HomePage. When I now try to do another relative navigation anywhere including to the "MyDetailsPage", no navigation happens and everything just seems stuck/frozen.

Note that same code was working fine on Prism.Unity.Forms v8.1.97.

Has anyone else seen this behaviour/bug? Are there any breaking changes to the navigation process in the latest 9.0.271 version?

**Comment by dansiegel** [comment]
I don't believe there have been any changes to the NavigationService for Prism.Forms from 8.1 to 9.0. We really only even have Prism.Forms for 9.0 to make it easier for people to first update their app to Prism 9.0 before making the migration to MAUI. If you're getting some sort of pausing before in Forms the issue would be coming from Xamarin.Forms itself

**Reply by SandipAhluwalia** [reply]
Hi @dansiegel, the same navigation service code works perfectly in 8.1 without any pausing. The only change I made was to update Prism.Unity.Forms from 8.1.97 to 9.0.271-pre. So if Prism.Forms was only made for 9.0 to aid MAUI migration then do you think it is safe to just ignore this behaviour in our Xamarin.Forms app and go ahead to replace Prism.Unity.Forms v9.0.271-pre with Prism.Dryloc.Maui v9.0.271-pre, and then continue the migration to MAUI assuming the NavigationService for Prism.Dryloc.Maui v9.0.271-pre works as before?

**Reply by dansiegel** [reply]
if the behavior is present in MAUI there is more that can be done about it with MAUI as Microsoft isn't really maintaining Xamarin.Forms but is doing bug fixes for MAUI and any issues that come up that are Prism specific we are looking at as we can.

**Reply by SandipAhluwalia** [reply]
So I tested a similar navigation scenario in a new MAUI app using Prism 9.0.401-pre and it seems to work as expected so it just seems to be broken in Xamarin.Forms. We will go ahead with the migration to MAUI via Prism 9.0.401-pre and see how it goes. If all goes well then we will become Commercial Plus license holders.
